packages/expect/src/toThrowMatchers.ts

```diff
@@ -79,10 +79,14 @@ export const createMatcher = (
 ): MatcherFunction<[any]> =>
   function (received, expected): ExpectationResult {
     const options = {
+      ...this.matcherHintOptions,
       isNot: this.isNot,
       promise: this.promise,
     };
 
+    const receivedColor = options.receivedColor ?? RECEIVED_COLOR;
+    const expectedColor = options.receivedColor ?? EXPECTED_COLOR;
```

Comment:
We already have `this.utils.RECEIVED_COLOR` and `this.utils.EXPECTED_COLOR` already. I wonder if we should add the configured colors to `this.utils` a bit more cleanly, and then use that ourselves instead of checking withing every single matcher?

Reply:
See #13567 - if this PR makes sure to tweak matcher context instead of looking through `options` in every matcher, it _should_ in theory be easier for all matchers to use the correct colors.